src/Symfony/Component/Messenger/Transport/Serialization/Serializer.php

```diff
@@ -34,9 +35,9 @@ class Serializer implements SerializerInterface
     private $format;
     private $context;
 
-    public function __construct(SymfonySerializerInterface $serializer, string $format = 'json', array $context = array())
+    public function __construct(SymfonySerializerInterface $serializer = null, string $format = 'json', array $context = array())
     {
-        $this->serializer = $serializer;
+        $this->serializer = $serializer ?? static::create()->serializer;
```

Comment:
weird-ish... why not inline then :) i suggest to use `Serializer::create()` in tests :thinking:

Reply:
makes it easier to use, don't you think? the code is implementation detail.